synapse/types/__init__.py

```diff
@@ -352,22 +370,31 @@ class EventID(DomainSpecificString):
     "_-./=+" + string.ascii_lowercase + string.digits
 )
 
+MXID_HISTORICAL_LOCALPART_ALLOWED_CHARACTERS = set(string.printable.replace(":", ""))
+
 # Guest user IDs are purely numeric.
 GUEST_USER_ID_PATTERN = re.compile(r"^\d+$")
 
 
-def contains_invalid_mxid_characters(localpart: str) -> bool:
+def contains_invalid_mxid_characters(
+    localpart: str, allow_historical_mxids: Optional[bool] = False
+) -> bool:
     """Check for characters not allowed in an mxid or groupid localpart
 
     Args:
         localpart: the localpart to be checked
-        use_extended_character_set: True to use the extended allowed characters
+        allow_legacy_mxids: True to use the extended allowed characters
             from MSC4009.
```

Comment:
Oops looks like I forgot to remove this, seems you named it `allow_historical_mxids` though (and the comment itself needs updating).